rpc/client.go

```diff
@@ -250,7 +273,7 @@ func (c *Client) SupportedModules() (map[string]string, error) {
 
 // Close closes the client, aborting any in-flight requests.
 func (c *Client) Close() {
-	if c.isHTTP {
+	if c.scheme == HTTPConn {
```

Comment:
An alternative change would have been to hide this internaly, so instead of `c.isHTTP {`, you would have `c.isHTTP() {`